lib/jobs/get-catalog-values-job.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+var di = require('di');
+
+module.exports = getCatalogValuesJobFactory;
+di.annotate(getCatalogValuesJobFactory, new di.Provide('Job.Get.Catalog.Values'));
+di.annotate(getCatalogValuesJobFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Services.Waterline',
+    'Util',
+    '_'
+));
+
+function getCatalogValuesJobFactory(
+    BaseJob,
+    Logger,
+    waterline,
+    util,
+    _
+){
+    var logger = Logger.initialize(getCatalogValuesJobFactory);
+
+    /**
+     * This job finds one or more requested values from one or more specified catalogs.
+     * The catalog values are made available in the shared context in an object that 
+     * maps the catalog values to requester provided keys.
+     *
+     * Example requestedData array:
+     * [{
+     *   "source": "ohai",
+     *   "keys": {
+     *     "myIPAddrKeyName": "data.ipaddress"
```

Comment:
I personally don't have any specific use case for utilizing the other properties, but just thought it might be beneficial for someone to be able to access all the information in the property, especially the created/modified times. I am not as familiar with how others use the catalog data, so it may be unnecessary. I'm good with either route.